database/migrations/2024_08_24_160326_label2test_relationship_refactor.php

```diff
@@ -35,17 +35,15 @@ public function up(): void
                 ', [$i]);
             }
         } else {
-            // MySQL is a bit more finicky about large transactions, so update in batches of 100
-            $rows_changed = 1;
-            while ($rows_changed > 0) {
-                $rows_changed = DB::update('
+            for ($i = 0; $i < ceil($count / 10000); $i++) {
+                DB::update('
                     UPDATE label2test, build2test
                     SET label2test.testid = build2test.id
                     WHERE
                         build2test.buildid = label2test.buildid
                         AND build2test.outputid = label2test.outputid
                         AND label2test.testid IS NULL
-                    LIMIT 100
+                        AND build2test.buildid % ? = 0
```

Comment:
how is `$i` getting bound to the `?` here?

Reply:
Good catch.  Thanks!